Update mruby_config_composite.yml

joelvbernier opened the pull request:
This is to reflect the changes in https://github.com/HEXRD/hexrd/pull/454.

joelvbernier committed:
Update mruby_config_composite.yml

This is to reflect the changes in https://github.com/HEXRD/hexrd/pull/454.

psavery commented:
@joelvbernier I gave this a try with `pytest` and the `active-hkls` branch in the hexrd repo. I got this error. I guess we should look into what is different:

```python
    def test_generate_eta_ome_maps(example_repo_include_path,
                                   test_config,
                                   reference_eta_ome_maps):
        os.chdir(example_repo_include_path)
        eta_ome_maps = generate_eta_ome_maps(test_config, save=False)
        eta_ome_maps = to_eomap(eta_ome_maps)
    
        expected = test_utils.load(reference_eta_ome_maps)
        comparison = test_utils.Comparison(expected, eta_ome_maps)
>       assert comparison.compare()
E       assert False
E        +  where False = <bound method Comparison.compare of <find_orientations_testing.Comparison object at 0x7f3f0de3b3d0>>()
E        +    where <bound method Comparison.compare of <find_orientations_testing.Comparison object at 0x7f3f0de3b3d0>> = <find_orientations_testing.Comparison object at 0x7f3f0de3b3d0>.compare

/home/patrick/virtualenvs/hexrd/src/hexrd/tests/test_find_orientations.py:110: AssertionError
```

psavery commented:
It looks like it was only failing because the `iHKLList` in the reference data did not match (since we changed that to be a master list instead of a reduced list).

I am about to push a commit that fixes the reference data to have the correct `iHKLList`.

psavery committed:
Fix reference data with changed/added features

Three keys were changed in the reference data: `iHKLList`, and
`planeData_excl`, `planeData_hkls`.

In the corresponding hexrd branch, the `iHKLList` in the eta omega maps
now refers to the master hkl indices rather than the reduced hkl indices.
Thus, we need to update the reference data to use that as well.

The `planeData_excl` is a newly added key, and we needed to add that here
as well.

The `planeData_hkls` now has the full hkl list and is modified by the
exclusions, rather than having a reduced hkl list.

These changes seem to fix the test.

Signed-off-by: Patrick Avery <patrick.avery@kitware.com>

psavery commented:
I fixed that issue, but now there is another issue. Here's the error I get:

```python
reference_orientations = array([[ 0.90291696,  0.16653558,  0.39615987, -0.00801406],
       [ 0.69819862,  0.54536815,  0.31252786,  0.34266399],
       [ 0.74318779,  0.64363269, -0.06431112, -0.17109339]])

    def test_find_orientations(example_repo_include_path,
                               example_repo_config_with_eta_ome_maps,
                               reference_orientations):
    
        os.chdir(example_repo_include_path)
        results = find_orientations(
             example_repo_config_with_eta_ome_maps
        )
    
        orientations = results['qbar']
    
        try:
            test_utils.compare_quaternion_lists(orientations.T,
                                                reference_orientations)
        except RuntimeError as err:
>           pytest.fail(str(err))
E           Failed: Incorrect number of orientations found; should be 3, currently found 0
```

psavery commented:
Okay! It appears to be working now. I modified `iHKLList` and `planeData_hkls` to be the full lists, and I added `planeData_excl`, and it appears to be passing the tests now.
